lib/identity_cache/configuration_dsl.rb

```diff
@@ -199,7 +199,7 @@ def identity_cache_single_value_dynamic_fetcher(fields, values) # :nodoc:
         cache_key = rails_cache_index_key_for_fields_and_values(fields, values)
         id = IdentityCache.fetch(cache_key) { identity_cache_conditions(fields, values).limit(1).pluck(primary_key).first }
```

Comment:
This would always store just an id in the cache.  Did you encounter an Array from adding a `unique: true` option to an existing cache_index?